src/cls/IPM/Storage/UpdateStep.cls

```diff
@@ -0,0 +1,102 @@
+Class %IPM.Storage.UpdateStep Extends %Persistent
+{
+
+Property ClassName As %Dictionary.Classname [ Required ];
+
+Property MethodName As %Dictionary.Identifier [ Required ];
+
+Property HostName As %String(MAXLEN = 255) [ Required ];
+
+Property InstanceName As %String(MAXLEN = 255) [ Required ];
+
+Index UpdateClassMethodIdx On (ClassName, MethodName, HostName, InstanceName) [ Unique ];
+
+Property Mirrored As %Boolean;
+
+Property TimeStampStart As %Library.DateTime;
+
+Property TimeStampEnd As %Library.DateTime;
+
+Property Status As %Status;
+
+Relationship ModuleHome As %IPM.Storage.Module [ Cardinality = one, Inverse = UpdateSteps, OnDelete = cascade ];
+
+Method %OnNew(
+	className As %String = "",
+	methodName As %String = "",
+	mirrored As %Boolean = 1,
+	hostName As %String = "",
+	instanceName As %String = "") As %Status
+{
+    set ..ClassName = className
+    set ..MethodName = methodName
+    set ..Mirrored = mirrored
+    if hostName = "" {
+        set ..HostName = ##class(%SYS.System).GetNodeName()
```

Comment:
Rather than doing this here, could simply have this be InitialExpression for HostName and InstanceName properties